signac/contrib/project.py

```diff
@@ -1397,8 +1438,8 @@ def _read_cache(self):
             logger.debug("Read cache in {:.3f} seconds.".format(delta))
             return cache
 
-    def index(self, formats=None, depth=0,
-              skip_errors=False, include_job_document=True):
+    def index_from_workspace(self, formats=None, depth=0,
+                             skip_errors=False, include_job_document=True):
```

Comment:
I think we should modify the `index()` method such that it calls both `index_from_workspace()` and `index_from_db()` and make the latter two private methods.

Reply:
That might be a good option. Are you proposing to use the union of the two sets of statepoints then? There could be statepoints that are only local, statepoints that are in the db, and state points that are in both.. need to make the latter unique.